src/backends/r1cs/mod.rs

```diff
@@ -463,6 +467,21 @@ where
             witness[var.index] = val;
         }
 
+        // print out the log info
+        for (_, span, var_info) in &self.log_info {
+            for cvar in var_info.var.iter() {
+                match cvar {
+                    ConstOrCell::Const(cst) => {
+                        println!("span:{}, cst: {}", span.start, cst.pretty());
+                    }
+                    ConstOrCell::Cell(cell) => {
+                        let val = cell.evaluate(&witness);
+                        println!("span:{}, val: {}", span.start, val.pretty());
+                    }
+                }
+            }
+        }
```

Comment:
I guess that works! So IIUC we just log the vars that we want to print out in order, and then when we generate the witness we go through that.

Maybe it would be worth moving this in the next loop so that we stop printing these if we hit an error in the constraint satisfaction

Reply:
> I guess that works! So IIUC we just log the vars that we want to print out in order, and then when we generate the witness we go through that.

Yes :)

This loop is not symmetric with the next loop. So atm it is not straightforward to merge these two loop. Maybe we don't need this once we got the IR.